flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestValuesTableFactory.java

```diff
@@ -145,7 +157,9 @@ private static RowKind parseRowKind(String rowKindShortString) {
 	// Factory
 	// --------------------------------------------------------------------------------------------
 
-	public static final String IDENTIFIER = "values";
+	public static final AtomicInteger RESOURCE_COUNTER = new AtomicInteger();
+
```

Comment:
What is this used for ?

Reply:
It is used for checking `open` and `close` are both invocked. The `RESOURCE_COUNTER` will be checked in https://github.com/apache/flink/pull/12047/files#diff-46716d3945346195315abf04aa338171R79

Reply:
Thanks for the explanation.